includes/tracks/class-wc-tracks-event.php

```diff
@@ -49,10 +49,17 @@ public function __construct( $event ) {
 	/**
 	 * Record Tracks event
 	 *
-	 * @return bool|WP_Error True on success, WP_Error on failure.
+	 * @return bool Always returns true.
 	 */
 	public function record() {
-		return WC_Tracks_Client::record_event( $this );
+		if (
+			( defined( 'DOING_AJAX' ) && DOING_AJAX ) ||
+			( defined( 'REST_REQUEST' ) && REST_REQUEST )
```

Comment:
On some initial testing, the `update_product` is not fired from PHP nor from a client side pixel when I update a product.

Is there another check missing from this if statement?

Reply:
I don't know if it would ever happen but you could do a check for did_action and `admin_footer` to avoid defering events too late.

Reply:
Fixed in 262afba - I believe `wp_redirect()` was the culprit for these events not being tracked (no `admin_footer`).